src/Rules/Library/NameExcludesSpecialCharacters.cs

```diff
@@ -22,8 +22,8 @@ public NameExcludesSpecialCharacters()
 
         public override EvaluationCode Evaluate(IA11yElement e)
         {
-            if (e == null) throw new ArgumentException(nameof(e));
-            if (e.Name == null) throw new ArgumentException(nameof(e.Name));
+            if (e == null) throw new ArgumentNullException(nameof(e));
+            if (e.name == null) throw new ArgumentException(ErrorMessages.ElementNameNullOrWhiteSpace, nameof(e));
```

Comment:
Looks like you grabbed my first version, which had a typo:
```suggestion
            if (e.Name == null) throw new ArgumentException(ErrorMessages.ElementNameNullOrWhiteSpace, nameof(e));
```